### PR TITLE
Fix: Spaced Comment Exceptions Not Working (Fixes #3276)

### LINKED ISSUE #3276: Spaced Comment Exceptions Not Working
I have the following config

```
"spaced-comment": [2, "always", { exceptions: ["*", "-"] }],
```

I'm trying the following code:

```
/**************
 * Private API
 **************/
```

It's giving me the following error:

```
models/abcModel.es6
  12:1  error  Expected exception block, space or tab after /* in comment  spaced-comment
```

I would assume that this would  **not warn**, because the docs give a similar example. Maybe it's an issue with `*` or something....

### PR DESCRIPTION
**Pull request by xjamundx**
Can someone spot check my logic here?

Assuming it's fine. I'll also update some docs and things before I'm done.

### CODE REVIEW

File: lib/rules/spaced-comment.js [hunk 1]
```diff
@@ -111,9 +113,8 @@ function createAlwaysStylePattern(markers, exceptions) {
             pattern += exceptions.map(escapeAndRepeat).join("|");
             pattern += ")";
         }
-        pattern += "$)"; // the sequence continues until the end.
+        pattern += (ending + ")"); // the sequence continues until the end.
```
Comment (xjamundx) [inline lib/rules/spaced-comment.js]: @mysticatea does this seem like a reasonable solution in your opinion?

Comment (mysticatea) [inline lib/rules/spaced-comment.js]: I thought `(?:$|[\n\r]))` is better :)

Comment (xjamundx) [inline lib/rules/spaced-comment.js]: I'll give it a shot, thought it may have side effects with multi-line line comments. let me check...

Comment (xjamundx) [inline lib/rules/spaced-comment.js]: You know your regexes! :+1:

**Comment by xjamundx** [comment]
Okay ready for review :)

File: lib/rules/spaced-comment.js (hunk 1 above)
Comment (mysticatea) [inline lib/rules/spaced-comment.js]: :+1:

File: lib/rules/spaced-comment.js [hunk 2]
```diff
@@ -111,9 +111,8 @@ function createAlwaysStylePattern(markers, exceptions) {
             pattern += exceptions.map(escapeAndRepeat).join("|");
             pattern += ")";
         }
-        pattern += "$)"; // the sequence continues until the end.
+        pattern += "(?:\$|[\n\r]))"; // the sequence continues until the end.
```
Comment (mysticatea) [inline lib/rules/spaced-comment.js]: oh, $ should not be escaped.

Comment (xjamundx) [inline lib/rules/spaced-comment.js]: but the tests pass .hahah. let's see what happens.

**Comment by nzakas** [comment]
LGTM

**Pull request merged**